common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java

```diff
@@ -210,10 +210,22 @@ public static List<Byte> stringToBytes(ExecutionContext ctx, Object str, String
         }
     }
 
-    private static byte[] bytesFromList(List<Byte> bytesList) {
+    private static byte[] bytesFromList(List<?> bytesList) {
         byte[] bytes = new byte[bytesList.size()];
         for (int i = 0; i < bytesList.size(); i++) {
-            bytes[i] = bytesList.get(i);
+            if (bytesList.get(i) instanceof Integer) {
+                byte val = ((Integer) bytesList.get(i)).byteValue();
+                if (((Integer) bytesList.get(i)).intValue() > 255 || ((Integer) bytesList.get(i)).intValue() < -128){
+                    throw new NumberFormatException("The value " + bytesList.get(i) + " could not be converted to a byte. Integer to byte needs only 8-bits (min/max = -128/127 or 0/255) !");
+                } else {
+                    bytes[i] = val;                }
+            } else if (bytesList.get(i) instanceof String) {
+                bytes[i] = parseInt((String) bytesList.get(i)).byteValue();
+            } else if (bytesList.get(i) instanceof Byte) {
+                bytes[i] = (byte) bytesList.get(i);
+            } else {
+                throw new NumberFormatException("Value \"" + bytesList.get(i) + " could not be converted to a byte. Must be format HexDecimal/String/Integer !");
```

Comment:
Please work with Gleb to make the error message more simple and clear.

Reply:
Ok